Scripts/HW_OOP.py

```diff
@@ -0,0 +1,483 @@
+# + Должен иметь метод `show_chat`, выводящий на экран все сообщения (каждое сообщение в таком же виде как и `show_last_message`, но с разделителем между ними)
+# + Должен иметь метод `recieve`, который будет принимать сообщение и добавлять его в чат
+# 
+# **Message**:
+# + Должен иметь три обязательных атрибута
+#     + `text` - текст сообщения
+#     + `datetime` - дата и время сообщения (встроенный модуль datetime вам в помощь). Важно! Это должна быть не дата создания сообщения, а дата его попадания в чат! 
+#     + `user` - информация о пользователе, который оставил сообщение (какой тип данных использовать здесь, разберётесь сами)
+# + Должен иметь метод `show`, который печатает или возвращает информацию о сообщении с необходимой информацией (дата, время, юзер, текст)
+# + Должен иметь метод `send`, который будет отправлять сообщение в чат
+# 
+# **User**:
+# + Класс с информацией о юзере, наполнение для этого класса придумайте сами
+# 
+# Напишите несколько примеров использования кода, которое показывает взаимодействие между объектами.
+# 
+# В тексте задания намерено не указано, какие аргументы должны принимать методы, пускай вам в этом поможет здравый смысл)
+# 
+# В этом задании не стоит флексить всякими продвинутыми штуками, для этого есть последующие
+# 
+# В этом задании можно использовать только модуль `datetime`
+
+
+import datetime
+
+
+class Chat:
+    def __init__(self, chat_history, chat_history_time_period):
+        self.chat_history = chat_history
+        self.chat_history_time_period = chat_history_time_period
+        
+    def show_last_message(self):
+        self.chat_history[0].show()
+    
+    def get_history_from_time_period(self, first_date, last_date):
+        for message in range(len(self.chat_history)):
+            if (self.chat_history[message].datetime > first_date) and (self.chat_history[message].datetime < last_date):
+                self.chat_history_time_period.insert(0, self.chat_history[message])
+                self.chat_history[message].show()
+                print("\n–––––––––––\n")
+        return self.chat_history_time_period
+    
+    
+    def show_chat(self):
+        for message in self.chat_history:
+            message.show()
+            print("\n–––––––––––\n")
+    
+    def recieve(self, message):
+        message.send()
+        self.chat_history.insert(0, message) 
+    
+
+
+class Message:
+    def __init__(self, text, datetime, user):
+        self.text = text
+        self.datetime = datetime
+        self.user = user
+        
+    def show(self):
+        showed_message = f'Message: {self.text}, \nDate and time: {self.datetime}, \nUser: {self.user}'
+        print(showed_message)
+
+
+    def send(self):
+        self.datetime = datetime.datetime.now()
+        return self
```

Comment:
Метод называется `send`, но send куда? Не понятно. По идее отправка должна происходить в чат
```suggestion
    def send(self, chat):
        self.datetime = datetime.datetime.now()
        chat.recieve(self)
```